lambdas/tests/unit/services/test_bulk_upload_metadata_service.py

```diff
@@ -0,0 +1,241 @@
+import tempfile
+from unittest.mock import call
+
+import pytest
+from botocore.exceptions import ClientError
+from models.staging_metadata import METADATA_FILENAME
+from pydantic import ValidationError
+from services.bulk_upload_metadata_service import BulkUploadMetadataService
+from tests.unit.conftest import (MOCK_LG_METADATA_SQS_QUEUE,
+                                 MOCK_LG_STAGING_STORE_BUCKET)
+from tests.unit.helpers.data.bulk_upload.test_data import (
+    EXPECTED_PARSED_METADATA, EXPECTED_SQS_MSG_FOR_PATIENT_1234567890,
+    EXPECTED_SQS_MSG_FOR_PATIENT_1234567891)
+
+MOCK_METADATA_CSV = "tests/unit/helpers/data/bulk_upload/metadata.csv"
+MOCK_INVALID_METADATA_CSV_FILES = [
+    "tests/unit/helpers/data/bulk_upload/metadata_invalid.csv",
+    "tests/unit/helpers/data/bulk_upload/metadata_invalid_empty_nhs_number.csv",
+    "tests/unit/helpers/data/bulk_upload/metadata_invalid_unexpected_comma.csv",
+]
+MOCK_TEMP_FOLDER = "tests/unit/helpers/data/bulk_upload"
+
+
+def test_process_metadata_send_metadata_to_sqs_queue(
+    set_env, mocker, metadata_filename, mock_sqs_service, mock_download_metadata_from_s3
+):
+    mock_download_metadata_from_s3.return_value = MOCK_METADATA_CSV
+    mocker.patch("uuid.uuid4", return_value="123412342")
+
+    service = BulkUploadMetadataService()
+    service.process_metadata(metadata_filename)
+
+    assert mock_sqs_service.send_message_with_nhs_number_attr_fifo.call_count == 2
+
+    expected_calls = [
+        call(
+            group_id="bulk_upload_123412342",
+            queue_url=MOCK_LG_METADATA_SQS_QUEUE,
+            message_body=EXPECTED_SQS_MSG_FOR_PATIENT_1234567890,
+            nhs_number="1234567890",
+        ),
+        call(
+            group_id="bulk_upload_123412342",
+            queue_url=MOCK_LG_METADATA_SQS_QUEUE,
+            message_body=EXPECTED_SQS_MSG_FOR_PATIENT_1234567891,
+            nhs_number="1234567891",
+        ),
+    ]
```

Comment:
Can we move this expected up to have the test in a Arrange-Act-Assert pattern?

Reply:
Of course!  Done just now :D